agent/acs/handler/task_manifest_handler.go

```diff
@@ -0,0 +1,280 @@
+	err := taskManifestHandler.acsClient.MakeRequest(&ecsacs.AckRequest{
+		Cluster:           aws.String(taskManifestHandler.cluster),
+		ContainerInstance: aws.String(taskManifestHandler.containerInstanceArn),
+		MessageId:         aws.String(messageID),
+	})
+	if err != nil {
+		seelog.Warnf("Error 'ack'ing TaskManifestMessage with messageID: %s, error: %v", messageID, err)
+	}
+}
+
+func (taskManifestHandler *taskManifestHandler) sendTaskTerminationVerificationMessage(
+	message *ecsacs.TaskStopVerificationMessage) error {
+
+	seelog.Debugf("Sending Task Termination Verification Message: %v", message)
+	err := taskManifestHandler.acsClient.MakeRequest(message)
+	if err != nil {
+		seelog.Warnf("Error 'ack'ing request with messageID: %s, error: %v", message.MessageId, err)
+		return err
+	}
+	return nil
+}
+
+// stop is used to invoke a cancellation function
+func (taskManifestHandler *taskManifestHandler) stop() {
+	taskManifestHandler.cancel()
+}
+
+func (taskManifestHandler *taskManifestHandler) handleTaskManifestMessage() {
+	for {
+		select {
+		case <-taskManifestHandler.ctx.Done():
+			return
+		case message := <-taskManifestHandler.messageBufferTaskManifest:
+			if err := taskManifestHandler.handleTaskManifestSingleMessage(message); err != nil {
+				seelog.Warnf("Unable to handle taskManifest message [%s]: %v", message.String(), err)
+			}
+		}
+	}
+}
+
+func (taskManifestHandler *taskManifestHandler) sendTaskStopVerificationMessage() {
+	for {
+		select {
+		case message := <-taskManifestHandler.messageBufferTaskStopVerificationMessage:
+			if err := taskManifestHandler.sendTaskTerminationVerificationMessage(message); err != nil {
+				seelog.Warnf("Unable to handle taskStopVerification message [%s]: %v", message.String(), err)
+			}
+		case <-taskManifestHandler.ctx.Done():
+			return
+		}
+	}
+}
+
+// compares the list of tasks received in the task manifest message and tasks running on the the instance
+// It returns all the task that are running on the instance but not present in task manifest message task list
+func compareTasks(receivedTaskList []*ecsacs.TaskIdentifier, runningTaskList []*apitask.Task) []*ecsacs.TaskIdentifier {
+
+	tasksToBeKilled := make([]*ecsacs.TaskIdentifier, 0)
+	for _, runningTask := range runningTaskList {
+		// For every task running on the instance check if the task is present in receivedTaskList with the DesiredState
+		// of running, if not add them to the list of task that needs to be stopped
+		if runningTask.GetDesiredStatus() == apitaskstatus.TaskRunning {
+			taskPresent := false
+			for _, receivedTask := range receivedTaskList {
+				if *receivedTask.TaskArn == runningTask.Arn && *receivedTask.DesiredStatus == "RUNNING" {
+					// Task present, does not need to be stopped
+					taskPresent = true
+					break
+				}
+			}
+			if !taskPresent {
+				tasksToBeKilled = append(tasksToBeKilled, &ecsacs.TaskIdentifier{
+					DesiredStatus: aws.String("STOPPED"),
```

Comment:
why is the string used here instead of constant(taskstatus.Stopped?)?

Reply:
Added a string constant.